microdnf/commands/advisory/advisory_subcommand.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef MICRODNF_COMMANDS_ADVISORY_ADVISORY_SUBCOMMAND_HPP
+#define MICRODNF_COMMANDS_ADVISORY_ADVISORY_SUBCOMMAND_HPP
+
+#include "arguments.hpp"
+#include "microdnf/context.hpp"
+
+#include "libdnf/advisory/advisory_query.hpp"
+
+#include <libdnf-cli/session.hpp>
+
+#include <memory>
+#include <vector>
+
+
+namespace microdnf {
+
+
+class AdvisorySubCommand : public libdnf::cli::session::Command {
+public:
+    void run() override;
+
+protected:
+    virtual void process_and_print_queries(
+        Context & ctx, libdnf::advisory::AdvisoryQuery & advisories, libdnf::rpm::PackageQuery & packages) = 0;
+
+    std::unique_ptr<AdvisoryAvailableOption> available{nullptr};
+    std::unique_ptr<AdvisoryInstalledOption> installed{nullptr};
+    std::unique_ptr<AdvisoryAllOption> all{nullptr};
+    std::unique_ptr<AdvisoryUpdatesOption> updates{nullptr};
+    std::unique_ptr<AdvisorySpecArguments> advisory_specs{nullptr};
+
+    explicit AdvisorySubCommand(Command & parent, const std::string & name, const std::string & short_description);
```

Comment:
You don't need to declare constructors with more than one argument as `explicit`, it only matters for one-argument constructors.